### PR TITLE
Porting to python3.5

### PR DESCRIPTION
**Pull request by navneeth**
I had the make the following changes to get the code to work with python3.5.
Please review it.

Currently I am stuck with the following error on json serialization:

The stack trace is the following.

  File "/Curve/api/curve/v1/validators.py", line 31, in default
    return json.JSONEncoder.default(self, o)
  File "/Curve/venv/lib/python3.5/site-packages/flask/json/__init__.py", line 81, in default
    return _json.JSONEncoder.default(self, o)
  File "/usr/lib/python3.5/json/encoder.py", line 179, in default
    raise TypeError(repr(o) + " is not JSON serializable")

Any pointer to fix it is appreciated. 
	
Next, despite using the sample csv shown in the README.md. The application does not load it.

Thanks !

### CODE REVIEW

File: api/curve/v1/__init__.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def auth_filter():
         g.user = User.query.filter_by(uid=uid).first()
     if g.user is None:
         login_path = path_prefix + '/login/redirect'
-        if request.path.encode('utf-8').startswith(path_prefix + '/login'):
+        if request.path.encode('utf-8').startswith( (path_prefix + '/login').encode('utf-8') ):
```
Comment (navneeth) [inline api/curve/v1/__init__.py]: File "/Curve/api/curve/v1/__init__.py", line 58, in auth_filter
    if request.path.encode('utf-8').startswith(path_prefix + '/login'):
TypeError: startswith first arg must be bytes or a tuple of bytes, not str

This fix is for the above error

File: api/curve/v1/utils.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def encode_if_unicode(s):
     :param s:
     :return:
     """
-    if isinstance(s, unicode):
+    if isinstance(s, str):
```
Comment (navneeth) [inline api/curve/v1/utils.py]: File "Curve/api/curve/v1/utils.py", line 80, in encode_if_unicode
    if isinstance(s, unicode):
NameError: name 'unicode' is not defined

Fix for above error

File: api/curve/v1/services/plugin.py [hunk 3]
```diff
@@ -206,7 +209,10 @@ def __call__(self, method, *args):
             if not isinstance(plugin.__dict__[method], types.FunctionType):
                 continue
             try:
+                current_app.logger.info('Invoking call with __call__ %s %s', plugin, method)
                 output = self._invoke(plugin, method, *args)
+                current_app.logger.info('Returned %s', output)
+                output = 'default', (2300, 2900)
```
Comment (minorcong) [inline api/curve/v1/services/plugin.py]: What's the advantage for this change? This line will cause an internal error during upload.

File: api/curve/v1/api/data_dataName.py [hunk 4]
```diff
@@ -82,15 +83,17 @@ def post(self, dataName):
...
         try:
             points, time_formatter = self._parse_file(upload_file)  # parse data in csv
         except Exception as e:
-            return self.render(msg=e.message, status=422)
+            return self.render(msg=str(e), status=422)
         if len(points) < 2:
             return self.render(msg='at least 2 point', status=422)
         timestamps = np.asarray(sorted(points.keys()))
         periods = np.diff(timestamps)
+        #current_app.logger.info('Periods as numpy %s', points.keys())
```
Comment (minorcong) [inline api/curve/v1/api/data_dataName.py]: It's better to delete the personal debug code.

File: api/curve/v1/plugins/axis_demo/__init__.py [hunk 5]
```diff
@@ -21,6 +22,7 @@ def y_axis(api, line):
     :param line: tuple-like ((timestamp, value)), the timestamp and value is const
     :return: plugin_name, (y_axis_min, y_axis_max)
     """
+    current_app.logger.info('Invoking y_axis plugin on  %s', line)
```
Comment (minorcong) [inline api/curve/v1/plugins/axis_demo/__init__.py]: This line will lead to a huge log file

File: api/curve/v1/utils.py (hunk 2 above)
Comment (minorcong) [reply api/curve/v1/utils.py]: Thanks for your revision.

File: api/curve/v1/__init__.py (hunk 1 above)
Comment (minorcong) [reply api/curve/v1/__init__.py]: Thanks for your revision.

File: api/curve/v1/services/plugin.py (hunk 3 above)
Comment (navneeth) [reply api/curve/v1/services/plugin.py]: Apologies. This was for debugging. I'll revise the PR.

Comment (navneeth) [reply api/curve/v1/services/plugin.py]: I have corrected this in the new PR.

File: api/curve/v1/api/data_dataName.py (hunk 4 above)
Comment (navneeth) [reply api/curve/v1/api/data_dataName.py]: I have corrected this in the new PR.

File: api/curve/v1/plugins/axis_demo/__init__.py (hunk 5 above)
Comment (navneeth) [reply api/curve/v1/plugins/axis_demo/__init__.py]: I have corrected this in the new PR.

**Pull request closed without merging**